File: examples/java/patterns/idempotency/idempotency-tokens.java
String idempotencyKey = context.step(
    "idempotency-key",
    String.class,
    ctx -> UUID.randomUUID().toString());

Receipt receipt = context.step(
    "charge",
    Receipt.class,
    ctx -> paymentService.charge(
        input.amount(), input.cardToken(), idempotencyKey));

File: examples/typescript/patterns/idempotency/idempotency-tokens.ts
const idempotencyKey = await context.step("idempotency-key", async () =>
  randomUUID(),
);

const charge = await context.step("charge", async () => {
  return paymentService.charge({
    amount: event.amount,
    cardToken: event.cardToken,
    idempotencyKey,
  });
});
